functions/zram.bash

```diff
@@ -74,6 +74,10 @@ init_zram_mounts() {
     if ! cond_redirect install -m 644 "$zramInstallLocation"/openhabian-zram/ro-root.sh /usr/local/share/zram-config/ro-root.sh; then echo "FAILED (ro-root)"; return 1; fi
     if cond_redirect install -m 644 "$zramInstallLocation"/openhabian-zram/zram-config.logrotate /etc/logrotate.d/zram-config; then echo "OK"; else echo "FAILED (logrotate)"; return 1; fi
 
+    if [[ -f /etc/systemd/system/find3server.service ]]; then
```

Comment:
That won't start the dir as ZRAM, only on next reboot.
If we want it right away as I said we need a one-time zram-config run for the single new dir like so
mv /etc/ztab /tmp
echo "...">/etc/ztab
/usr/local/bin/zram-config start
mv /tmp/ztab /etc

Reply:
That is before ZRAM has been initialized so it should be fine. This is in the ZRAM install script.

Reply:
that won't work.
it is in ZRAM *install* script which is run at install time only. FIND install is not run.
FIND will be added later, interactively

Reply:
It is also run from the menu if a user decides to uninstall and reinstall.

Reply:
I don't think it makes sense that way:
On unattended install, the find3server unit file is not there thus nothing is added to ztab
When a user decides to install FIND, it won't (and shouldn't) call the zram install routine, will it?
And even if we did, it won't start the new ZRAM dir without a reboot.
So better use my sniplet.

Reply:
No, the code to add it appears in both places so that it will be added in both cases. Check find.bash again to see what I mean.

Reply:
ok, it's added. But the point is it still is not activated.
That's why you need to run zram-config start with a one-line ztab. Or reboot, but that's not necessary.

Reply:
Yes it will be, it is activated in both. In `zram.bash` the line is added to `/etc/ztab` before ZRAM is started for the first time which means it will automatically be added after the first start.

In `find.bash` it is activated using a `zram-config start` command directly following it. I see no issues with this. 

Quite frankly, I prefer the original method which was simply stopping zram and restarting because that is the recommend method to do it anyway. It shouldn't matter if openhab is restarted because this will only be run during server maintenance or setup to begin with.

Reply:
> In `find.bash` it is activated using a `zram-config start` command directly following it. I see no issues with this.

zram-config isn't built to cope with changes to ztab. It duplicates at least some 'log' ztab lines resulting in extra zram devices we don't want. The method to prevent is to use a temporary one-line ztab.
ZRAM restarts are ugly (stops syslog, does lazy unmounting and probably has more potential breaking issues) so I want to avoid that.

On a similar note, we also need to ensure we don't add to ztab more than once e.g. if a user runs FIND install twice.

Reply:
Perhaps a better solution is to change the upstream ZRAM code to be safer and more reliable? IMHO if a service (in this case ZRAM) is intended to be used in a specific way (in this case stop and restart to allow for safe and proper handling) it should function properly in that way and the end user should not have to worry about it.

I personally have never observed any issues with restarting ZRAM and it has always been fairly quick in my experience. I would prefer to use the service method because it avoids the duplication of ZRAM which could lead to not enough RAM being available, and it avoids the messy process of adding to a live ZRAM config.

Because of this, if you don't mind I would prefer to stick with starting and stopping the service and if issues arise we can look into using your method.

Reply:
> Perhaps a better solution is to change the upstream ZRAM code to be safer and more reliable?

Yes but that's tough. It was written by someone who no longer maintains it and I've taken maintainership and it took me many hours to reverse engineer but the code really is not built to handle that, it would need a major refactoring which I'm not really keen to take on.

> I personally have never observed any issues with restarting ZRAM

But I have, and I've spent countless hours supporting others that did.
Just spent many more with a non-UNIX guy (duh!) and yes, sometimes even after zram stop there's a ZRAM device left.
I agree that we must avoid duplication (not sure if it really uses RAM twice but a fair chance it does) but I hence disagree service stop is a better or safe way.
This is exactly why I've put testing efforts into this and came up with that other method so I really prefer that.

Reply:
Very well, I have added it.